functions/index.ts

```diff
@@ -0,0 +1,168 @@
+import { Game as Game, IHttpResponse as IHttpResponse } from "./Modules";
+const allowedRequestParameters = JSON.parse(process.env.ALLOWED_REQUEST_PARAMETERS!);
+
+exports.handler = async (event: any, context: any, callback: any) => {
+  switch (event.path) {
+    case ("/getGame"):
+      await getGame(event.queryStringParameters["gameName"]);
+      break;
+    case ("/listGames"):
+      await listGames();
+      break;
+    case("/createGame"):
+      await createGame(new Game(JSON.parse(event.body).gameName, JSON.parse(event.body).yearReleased, JSON.parse(event.body).genre, JSON.parse(event.body).console, JSON.parse(event.body).developer));
+      break;
+    case("/modifyGame"):
+      await modifyGame(new Game(JSON.parse(event.body).gameName, JSON.parse(event.body).yearReleased, JSON.parse(event.body).genre, JSON.parse(event.body).console, JSON.parse(event.body).developer));
+      break;
+    case ("/deleteGame"):
+      await deleteGame(JSON.parse(event.body).gameName);
+      break;
+    default:
+      HttpResponse({statusCode: 400, body: JSON.stringify("Unable to process request")});
+      break;
+  }
+  
+  async function getGame(gameName: string) {
+    let params = {
+      TableName: table,
+      Key: {
+        gameName: gameName
+      }
+    };
+
+    try {
+      let response = await docClient.get(params).promise();
+      if (response.Item) {
+        let game = new Game(response.Item.gameName, response.Item.yearReleased, response.Item.genre, response.Item.console, response.Item.developer);
+        HttpResponse({statusCode: 200, body: JSON.stringify(game)});
+      } else {
+        throw Error;
+     }
+    } catch (err) {
+      throw HttpResponse({statusCode: 404, body: `Unable to find game.`});
+    }
+  }
+  
+  async function listGames() {
+    let params = {
+      TableName: table,
+      Select: "ALL_ATTRIBUTES"
+    };
+    
+    try {
+      let response = await docClient.scan(params).promise();
+      let gameList : Game[] = [];
+      
+      if (response.Items) {
+        response.Items.forEach((game: Game) => {
+          gameList.push(new Game(game.gameName, game.yearReleased, game.genre, game.console, game.developer));
+        })
+        HttpResponse({statusCode: 200, body: JSON.stringify(gameList)})
+      } else {
+        throw Error;
+     }
+    } catch (err) {
+      throw HttpResponse({statusCode: 404, body: `Game list could not be found.`});
+    }
+  }
+
+  async function createGame(game: Game) {
+    let params = {
+      TableName: table,
+      Item: {
+        gameName: game.gameName,
+        genre: game.genre,
+        yearReleased: game.yearReleased,
+        developer: game.developer,
+        console: game.console
+      },
+      ConditionExpression: 'attribute_not_exists(gameName)'
+    }
+
+    try {
+      let response = await docClient.put(params).promise();
+      if (Object.keys(response).length == 0) {
+        HttpResponse({statusCode: 201, body: JSON.stringify(game)});
```

Comment:
This is fine for now but in the future you might want to consider pulling the game out of the DB in case there are any generated attributes (like an ID) down the road.